tools/java_api/src/jni/kuzu_java.cpp

```diff
@@ -1119,6 +1132,26 @@ JNIEXPORT jlong JNICALL Java_com_kuzudb_Native_kuzu_1value_1get_1struct_1index(J
     }
 }
 
+JNIEXPORT jlong JNICALL Java_com_kuzudb_Native_kuzu_1value_1get_1map_1index(JNIEnv* env, jclass,
```

Comment:
I suggest to just implement a function to get value by index (similar to C API) and pass a `jlong index` instead of `jstring field_name`. Then, if iterate through the object to get by field name is required, the logic of iterating can be pushed to Java side.